### PR TITLE
Enum names Type and Type2.

### LINKED ISSUE #590: Enum names Type and Type2.
When using `npx @rtk-query/codegen-openapi` (which I think uses this project) and you have the following in `swagger.json`
```
{
...
  "definitions": {
    "Attachment": {
      ...
      "type": {
        "type": "string",
        "enum": [
          "picture",
          "document",
          ...
        ],
        "x-ts-type": "AttachmentType"
      },
    },
    ...
    "Company": {
      ...
      "type": {
        "type": "string",
        "enum": [
          "inc",
          ..
        ],
        "x-ts-type": "CompanyType"
      },
    },
...
}
```
Then codegen-openapi generates enums `Type`, `Type2` is there a way to make it generate `AttachmentType`  and `CompanyType`  instead?

### PR DESCRIPTION
**Pull request by 2Fast2BCn**
Attempt to fix #590

Made it a draft PR because I kinda doubt it would be this easy ;-) .

### CODE REVIEW

**Comment by Xiphe** [comment]
Hi @2Fast2BCn 

Thank you for bringing this forward. This might already be it if you ask me. Can you also add some test-case illustrating the expected behavior?

File: packages/codegen/src/generate.ts [hunk 1]
```diff
@@ -795,14 +796,18 @@ export default class ApiGenerator {
...
       .split(/[^A-Za-z0-9$_]/g)
       .map((n) => _.upperFirst(n))
       .join("");
     const stringEnumValue = this.getEnumValuesString(
       schema.enum ? schema.enum : [],
     );
 
+    const tsName = schema["x-ts-type"];
+
+    const proposedName = tsName ? tsName : autoGeneratedName;
```
Comment (Xiphe) [inline packages/codegen/src/generate.ts:809]: How about

```ts
const proposedName =
  schema["x-ts-type"] ??
  baseName
    .split(/[^A-Za-z0-9$_]/g)
    .map((n) => _.upperFirst(n))
    .join("");
```

File: packages/codegen/src/generate.ts [hunk 2]
```diff
@@ -795,14 +796,18 @@ export default class ApiGenerator {
     const baseName = schema.title || _.upperFirst(propName);
```
[the remark this answers is not in the document]
Comment (2Fast2BCn) [reply packages/codegen/src/generate.ts:796]: oh boy
OK I'll just trip my code changes and leave only the test.

[the remark this answers is not in the document]
Comment (2Fast2BCn) [reply packages/codegen/src/generate.ts]: This however doesn't fix my problem.
In swagger I cannot add title.
It has to be an extension and start with `x-`
So should we duplicate this or?

**Comment by 2Fast2BCn** [comment]
Closing the PR it's not the problem of this library.

**Comment by Xiphe** [comment]
Again: Sorry for the work you've put in this. And thanks a lot that we found that `title` just works!

**Pull request closed without merging**